dhis-2/dhis-services/dhis-service-administration/src/main/resources/data-integrity-checks/tracked_entity_attributes/indexable_tracked_entity_attributes_block_trigram_operators.yaml

```diff
@@ -0,0 +1,31 @@
+---
+name: indexable_tracked_entity_attributes_block_trigram_operators
+description: Tracked entity attributes that are trigram indexable but restrict the operators required for trigram-based searches.
+section: Tracked entity attributes
+section_order: 1
+summary_sql: >-
+    select count(*) as value, 100*count(*) / NULLIF((select count(*) from trackedentityattribute),0) as percent
+    from trackedentityattribute
+    where trigramindexable = true
+    AND (
+    	blockedsearchoperators @> CAST('["LIKE"]' AS jsonb)
+        AND blockedsearchoperators @> CAST('["EW"]' AS jsonb)
+    );
+details_sql: >-
+    select uid, name
+    from trackedentityattribute
+    where trigramindexable = true
+    AND (
+    	blockedsearchoperators @> CAST('["LIKE"]' AS jsonb)
+        AND blockedsearchoperators @> CAST('["EW"]' AS jsonb)
+    );
+details_id_type: trackedEntityAttributes
+severity: WARNING
+introduction: >
+    A trigram index can be created on a Tracked Entity Attribute (TEA) if the TEA has the trigram indexable flag set to true and it supports at least one of the operators: LIKE or EW.
+    That means a TEA with the flag set to true but blocking both operators, will never have a trigram index.
+    The TRACKER_TRIGRAM_INDEX_MAINTENANCE job in the scheduler app is responsible for creating these indexes.
+recommendation:
+    Consider allowing at least one of the LIKE or EW operators on the TEAs below.
+    This will enable the creation of a trigram index on the tracked entity value, improving search performance on tracked entities.
```

Comment:
```suggestion
    This will make sure the trigram indexable flag is enabled only for those Tracked Entity Attributes where it is useful to have the partial trigram index.
```